synapse/handlers/federation.py

```diff
@@ -292,38 +295,42 @@ async def _maybe_backfill_inner(
         # TODO: If we do do a backfill then we should filter the backwards
         #   extremities to only include those that point to visible portions of
         #   history.
-        #
-        # TODO: Correctly handle the case where we are allowed to see the
-        #   forward event but not the backward extremity, e.g. in the case of
-        #   initial join of the server where we are allowed to see the join
-        #   event but not anything before it. This would require looking at the
-        #   state *before* the event, ignoring the special casing certain event
-        #   types have.
-
-        forward_event_ids = await self.store.get_successor_events(
-            (e.event_id for e in backwards_extremities)
-        )
 
-        extremities_events = await self.store.get_events(
-            forward_event_ids,
-            redact_behaviour=EventRedactBehaviour.AS_IS,
-            get_prev_content=False,
-        )
+        found_filtered_extremity = False
```

Comment:
Part of me wonders if it might be slightly neater to set `filtered_extremities = []` rather than introduce a new boolean here. Not going to insist though (+0.2).

Reply:
Looking deeper, it doesn't seem like we actually use the list of `filtered_extremities` directly. Given that, I think this suggestion is unhelpful.